docs/design/proposals/carm/cross-account-resource-management.md

```diff
@@ -0,0 +1,177 @@
+
+#### Design details
+
+##### Namespace annotations
+
+To associate a namespace with a unique AWS account, ACK service controllers expect from cluster administrators to create namespaces with a specific annotation `services.k8s.aws/owner-account-id` e.g:
+
+```yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: production-n1
+  annotations:
+    services.k8s.aws/default-aws-region: "eu-west-1"
+    services.k8s.aws/owner-account-id: "123456789012"
+```
+
+Each time a new AWS resource object is updated, ACK service controllers will lookup the namespaces annotations and decide in which namespace the AWS object should be created.
+
+In addition to the owner account id annotation, users can also specify a `services.k8s.aws/default-aws-region` annotation. This region will be used by default in case region annotation is missing in the CRDs specs.
+
+*//TODO(a-hilaly) address namespace/account-id conflict cases*
+
+*//TODO(a-hilaly) address namespace deletion case*
+
+##### Region determination
+
+To determine within which region the resources should be created, ACK controllers will, in order, look for a region in the following sources: 
+
+- Resource region annotation `services.k8s.aws/aws-region`. If provided it will override the namespace default region annotation.
+- Namespace default region annotation `services.k8s.aws/default-aws-region`
+- If none of the two annotations are provided ACK will try to find a region from this sources: (see [cli user guide](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html#cli-configure-quickstart-precedence)): Controller flags, Pod IRSA environment variables...
+
+##### Storing AWS Role ARNs
+
+AccountIDs and their associate AWS Role ARNs will be stored in a `ConfigMap` called "ack-carm-config-map". e.g:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ack-carm-config-map
+  namespace: default
+data:
+  accounts: |
+    123456789012=arn:aws:iam::123456789012:root
+    454545454545=arn:aws:iam::454545454545:role/S3Access
+```
+
+##### IAM Account pivot
+
+When a matching namespace is found, ACK service controllers will search for the IAM Role ARN associated with the AccountID in the `ack-carm-config-map` ConfigMap, and pivot the session to create resources in a new AWS account.
+
+The pivot is done by using the [aws-sdk-go](https://github.com/aws/aws-sdk-go) `session` package. `session.NewSession()` function will load the AWS related environment variables (injected by Pod IRSA) and initialize the [token provider](https://github.com/aws/aws-sdk-go/blob/master/aws/session/session.go#L200-L218) function. Later the token provider function is executed when [`AssumeRoleProvider.Retrieve`](https://github.com/aws/aws-sdk-go/blob/master/aws/credentials/stscreds/assume_role_provider.go#L329) is called and [return a set of credentials](https://github.com/aws/aws-sdk-go/blob/master/aws/credentials/stscreds/assume_role_provider.go#L357-L362) to make pivoted api calls.
```

Comment:
Super! Feel free to fill in more here, including potentially some sample code snippets.